features/GEM005_geometry-decomposed.objects.feature

```diff
@@ -0,0 +1,31 @@
+@informal-proposition
+@gem
+Feature: GEM005 - Constraints on representation identifiers
+
+"""
+    In cases of aggregation of physical elements into a physical aggregate
+    the shape representation of the whole (within the same representation identifier)
+    must be taken from the sum of the shape representations of the parts.
+
+    If a building element acts as container, i.e. has parts associated,
+    and those parts have own shape representations, then the container
+    must have no shape representations other than 'Body'.
+"""
+  Background: Entity is part having own shape representation
+
+    Given An IfcBuildingElement
+      And A relationship IfcRelAggregates to IfcBuildingElement from IfcObject
+      And Its attribute Representation 
+      And Its attribute Representations 
+      And Its attribute RepresentationIdentifier and return to first
```

Comment:
I have some objections to this:
 - `return to first` is not very general as it can only jump back to the top of the stack
 - `return to first` seems to set `context.applicable` which is surprising behaviour
 - I think we need to specifically see if a body representation exists, but I'm not sure, the wording of the rule is not very precise.

Can we work towards something like this:

```suggestion
      And its attribute RepresentationIdentifier
      And the value is "Body"
      And return to IfcBuildingElement
```